### PR TITLE
vertx-redis: implement v1 naming and surface remote connection info

### PR DESCRIPTION
**Pull request by amarziali**
# What Does This Do

Implements opt-in v1 naming convestions:
* operation: redis.command
* service: DD_SERVICE

Surface remote host connection tags:
* peer.hostname
* peer.port

# Motivation

# Additional Notes

### CODE REVIEW

File: dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisAPIImplSendAdvice.java [hunk 1]
```diff
@@ -26,15 +32,25 @@ public static void afterSend(@Advice.This RedisAPI self, @Advice.Return Future<R
...
+        if (handler.clientSpan != null && connection != null) {
+          final SocketAddress socketAddress =
+              InstrumentationContext.get(RedisConnection.class, SocketAddress.class)
+                  .get(connection);
+          if (socketAddress != null) {
+            DECORATE.onConnection(handler.clientSpan, socketAddress);
+            DECORATE.setPeerPort(handler.clientSpan, socketAddress.port());
+          }
+        }
+        if (!future.isComplete()) {
```
Comment (ygree) [inline dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisAPIImplSendAdvice.java:45]: NIce!

File: dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/ConnectionAsyncResultWrapper.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    this.contextStore = contextStore;
+    this.redisInstance = redisInstance;
+  }
+
+  @Override
+  public void handle(AsyncResult<RedisConnection> event) {
+    delegate.handle(
+        event.map(
+            connection -> {
+              contextStore.put(redisInstance, connection);
```
Comment (ygree) [inline dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/ConnectionAsyncResultWrapper.java:30]: Here we associate one connection with a Redis client instance. Could there be more than one active connections for the same Redis client instance?

File: dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisAPICallAdvice.java [hunk 3]
```diff
@@ -134,7 +137,23 @@ public static boolean beforeCall(
         break;
       default:
     }
-
+    if (self instanceof RedisAPIImpl) {
+      Object connection =
+          InstrumentationContext.get(RedisAPIImpl.class, Object.class).get((RedisAPIImpl) self);
+      if (connection instanceof Redis) {
```
Comment (ygree) [inline dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisAPICallAdvice.java:143]: This two if checks are read as that the connection object associated with RedisAPIImpl can be either Redis or RedisConnection which is a bit confusing. If it's never the case and the object associated with RedisAPIImpl is always Redis then moving the next if-check inside of the first if-check scope would be less confusing.

Comment (ygree) [reply dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisAPICallAdvice.java:143]: The same applies to [RedisSendAdvice.java](https://github.com/DataDog/dd-trace-java/pull/5041/files#diff-8981ca5bfd751f6700ee29de7d46bce2444dd47b3cc34333da14bb796adb7b2b)

**Review by ygree — CHANGES_REQUESTED** [review]
Looks good! Just got a couple of questions

File: dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisAPIImplSendAdvice.java (hunk 1 above)
Comment (amarziali) [reply dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisAPIImplSendAdvice.java:45]: Thanks

File: dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisAPICallAdvice.java (hunk 3 above)
Comment (amarziali) [reply dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisAPICallAdvice.java]: Finally I found another approach so this is now unnecessary

File: dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/ConnectionAsyncResultWrapper.java (hunk 2 above)
Comment (amarziali) [reply dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/ConnectionAsyncResultWrapper.java]: Yes completely. You are right. I completely changed the approach in order to have something only depending to the connection (without relating to other objects) 
Now it looks to my eyes way better

File: dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisConnectAdvice.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import datadog.trace.bootstrap.instrumentation.api.AgentScope;
+import datadog.trace.bootstrap.instrumentation.api.AgentSpan;
+import net.bytebuddy.asm.Advice;
+
+public class RedisConnectAdvice {
+  @Advice.OnMethodEnter
+  public static AgentScope before() {
+    final AgentSpan span = activeSpan();
+    if (span != null && VertxRedisClientDecorator.REDIS_COMMAND.equals(span.getOperationName())) {
+      return activateSpan(span, true);
```
Comment (ygree) [inline dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisConnectAdvice.java:15]: I'm not sure why do we need this advice? RedisAPICallAdvice should already activate the command span scope.

**Review by ygree — APPROVED** [review]
Nice fix! The only thing I'm not sure is still needed is RedisConnectAdvice

File: dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisConnectAdvice.java (hunk 4 above)
Comment (amarziali) [reply dd-java-agent/instrumentation/vertx-redis-client-3.9/src/main/java/datadog/trace/instrumentation/vertx_redis_client/RedisConnectAdvice.java:15]: Actually when used in pooled mode `Redis#send` is calling itself `connect` in a lambda. Without activating in this way I do not have the span in the other advices. The solution I found is to activate here

**Pull request merged**